src/parser/mod.rs

```diff
@@ -618,6 +632,7 @@ impl<'a> Parser<'a> {
                 // `COMMENT` is snowflake specific https://docs.snowflake.com/en/sql-reference/sql/comment
                 Keyword::COMMENT if self.dialect.supports_comment_on() => self.parse_comment(),
                 Keyword::PRINT => self.parse_print(),
+                Keyword::GO => self.parse_go(),
```

Comment:
can we rewind with `self.prev_token()` here before calling parse_go? so that the `parse_go` function is self contained and can parse a full GO statement. (I also realise now we missed that for print)

Reply:
I see that some of the statement types in parse_statement are doing that but most aren't. What's the strategy on this? How do you know when you're supposed to do it or not?

Reply:
Indeed the code is inconsistent in that regard historically, for newer statements ideally the parser functions are self contained when possible

Reply:
Thanks, that makes sense. I've updated it to rewind before parsing (using raiserror as a reference)